src/test/java/uk/gov/hmcts/befta/data/JsonStoreHttpTestDataSourceTest.java

```diff
@@ -0,0 +1,145 @@
+        assertEquals(3, requestData.getQueryParams().keySet().size());
+        assertEquals(3, requestData.getBody().keySet().size());
+        assertEquals(3, requestData.getHeaders().keySet().size());
+        assertEquals("REQ PATH VAR 3", requestData.getPathVariables().get("reqvar3"));
+        assertEquals("REQ PARAM 3", requestData.getQueryParams().get("reqparam3"));
+        assertEquals("REQ BODY 3", requestData.getBody().get("reqbody3"));
+        assertEquals("REQ HEADER 3", requestData.getHeaders().get("reqheader3"));
+
+        assertEquals(400, responseData.getResponseCode());
+        assertEquals(3, responseData.getHeaders().keySet().size());
+        assertEquals(3, responseData.getHeaders().keySet().size());
+        assertEquals("RES BODY 3", responseData.getBody().get("resbody3"));
+        assertEquals("RES HEADER 3", responseData.getHeaders().get("resheader3"));
+    }
+
+    @Test
+    public void shouldGetInheritedDataForTestCallAndOverrideSuccessfully() {
+        HttpTestData result = dataSource.getDataForTestCall("Simple-Data-With-Overrides");
+        RequestData requestData = result.getRequest();
+        ResponseData responseData = result.getExpectedResponse();
+        UserData invokingUser = result.getUsers().get("invokingUser");
+
+        assertEquals("Simple-Data-With-Overrides", result.get_guid_());
+        assertEquals("NEW USERNAME", invokingUser.getUsername());
+        assertEquals("NEW PASSWORD", invokingUser.getPassword());
+        assertEquals("789", invokingUser.getId());
+        assertEquals(invokingUser, result.getInvokingUser());
+        assertEquals(1, result.getUsers().size());
+
+        assertEquals(2, requestData.getPathVariables().keySet().size());
+        assertEquals(2, requestData.getQueryParams().keySet().size());
+        assertEquals(2, requestData.getBody().keySet().size());
+        assertEquals(2, requestData.getHeaders().keySet().size());
+        assertEquals("NEW REQ PATH VAR 2", requestData.getPathVariables().get("reqvar2"));
+        assertEquals("NEW REQ PARAM 2", requestData.getQueryParams().get("reqparam2"));
+        assertEquals("NEW REQ BODY 2", requestData.getBody().get("reqbody2"));
+        assertEquals("NEW REQ HEADER 2", requestData.getHeaders().get("reqheader2"));
+
+        assertEquals(200, responseData.getResponseCode());
+        assertEquals(2, responseData.getHeaders().keySet().size());
+        assertEquals(2, responseData.getHeaders().keySet().size());
+        assertEquals("NEW RES BODY 2", responseData.getBody().get("resbody2"));
+        assertEquals("NEW RES HEADER 2", responseData.getHeaders().get("resheader2"));
+    }
+
+    @Test
+    public void shouldReturnNullWhenGettingDataForTestDataIdNotFound() {
+        HttpTestData result = dataSource.getDataForTestCall("Non-Existing-Id");
+        assertNull(result);
+    }
+
+    private void simpleDataAsserts(HttpTestData result) {
```

Comment:
the method name could follow an imperative style like applyCommonAssertionsOn(result).

Reply:
Moved some asserts against the key test files into new CommonAssertions class with new naming.